vertexai/context-caching/contextcaching_update.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// contextcaching shows an example of caching the tokens of a mulitple PDF prompt
+package contextcaching
+
+// [START generativeaionvertexai_gemini_update_context_cache]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// updateContextCache shows how to update the expiration time of a cached content, by specifying
+// a new TTL (time-to-live duration)
+func updateContextCache(w io.Writer, contentName string, projectID, location string) error {
+	// location := "us-central1"
+	ctx := context.Background()
+
+	client, err := genai.NewClient(ctx, projectID, location)
+	if err != nil {
+		return fmt.Errorf("unable to create client: %w", err)
+	}
+	defer client.Close()
+
+	cachedContent, err := client.GetCachedContent(ctx, contentName)
+	if err != nil {
+		return fmt.Errorf("GetCachedContent: %w", err)
+	}
+
+	update := &genai.CachedContentToUpdate{
+		Expiration: &genai.ExpireTimeOrTTL{TTL: 2 * time.Hour},
+	}
+
+	_, err = client.UpdateCachedContent(ctx, cachedContent, update)
+	fmt.Fprintf(w, "Update cached content %q", contentName)
```

Comment:
Why don't you print the content of cached content here, like its corresponding Python sample does?
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/621454ef68f7fbda02060d1a249acb83d5e490d6/generative_ai/context_caching/update_context_cache.py#L35

Reply:
We only have metadata, like "expire_time". To access it, we need to call `GetCachedContent` again after `UpdateCachedContent`.  This is not the main job of `updateContextCache`, so instead we're doing this consistency check in part (4) of `TestContextCaching`.